Foundation/UserDefaults_Arguments.swift

```diff
@@ -0,0 +1,57 @@
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+
+import CoreFoundation
+
+fileprivate let propertyListPrefixes: Set<Character> = [ "{", "[", "(", "<", "\"" ]
+
+internal extension UserDefaults {
+    static func _parseArguments(_ arguments: [String]) -> [String: Any] {
+        var result: [String: Any] = [:]
+        
+        let count = arguments.count
+        
+        var index = 0
+        while index < count - 1 { // We're looking for pairs, so stop at the second-to-last argument.
+            let current = arguments[index]
+            let next = arguments[index + 1]
+            if current.hasPrefix("-") && !next.hasPrefix("-") {
```

Comment:
I bet we have subtle unicode behavior handling differences due to Swift string vs NSString, but I don’t think that’s a problem we need to solve here, at least not right now

Reply:
As far as I can see here, this allows a subset of all defaults we allow on Darwin — namely, we skip all defaults that start with a '-' code point but where that code point forms a grapheme, e.g. '-⃣'. It still allows basically all defaults keys of actual interest, though. But yep.

Reply:
… all defaults _arguments_.